From 8485c1c6e0722d0ee4629f9f4fffc20701f071ac Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Sat, 28 Sep 2024 13:52:38 +0800
Subject: [PATCH 232/277] mtk: mt76: mt7990: disable PPDU TxS to prevent MCU
 command timeout

Do not send to upstream!
This is a temporary solution. Remove it after rootcause found.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt7996/init.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mt7996/init.c b/mt7996/init.c
index 362a4fc7..e8fc6676 100644
--- a/mt7996/init.c
+++ b/mt7996/init.c
@@ -588,7 +588,9 @@ mt7996_mac_init_band(struct mt7996_dev *dev, u8 band)
 	/* MT_TXD5_TX_STATUS_HOST (MPDU format) has higher priority than
 	 * MT_AGG_ACR_PPDU_TXS2H (PPDU format) even though ACR bit is set.
 	 */
-	mt76_set(dev, MT_AGG_ACR4(band), MT_AGG_ACR_PPDU_TXS2H);
+	/* FIXME: In mt7990, it would lead to command timeout */
+	if (!is_mt7990(&dev->mt76))
+		mt76_set(dev, MT_AGG_ACR4(band), MT_AGG_ACR_PPDU_TXS2H);
 
 	if (!is_mt7996(&dev->mt76))
 		mt7996_mcu_set_bssid_mapping_addr(&dev->mt76, band);
-- 
2.45.2

